pmm-app/tests/pages/homePage.js

```diff
@@ -13,68 +14,91 @@ module.exports = {
         "//span[@class='panel-title-text' and contains(text(), 'Monitored DB Instances')]//../../../..//span[@class='singlestat-panel-value']",
     dashboardHeaderText: 'Percona Monitoring and Management',
     dashboardHeaderLocator: "//div[contains(@class, 'dashboard-header')]",
-    checkUpdateButton: "#refresh",
     oldLastCheckSelector: "#pmm-update-widget > .last-check-wrapper p",
-    lastCheckSelector: ".last-check-wrapper > p",
-    triggerUpdate: "button[ng-click='update()']",
-    updateProgressModal: ".modal-content",
-    reloadButtonAfterUpgrade: "button[ng-click='reloadAfterUpdate()'",
-    pmmUpdateWidget: "#pmm-update-widget",
     whatsNewLink: "a.text-primary.pmm-link",
-    upToDateLocator: "//section[@class='state']/p[text()='You are up to date']",
-    availableVersion: "#available_version > div > p",
-    currentVersion: "#current_version > span",
     sttDisabledFailedChecksPanelSelector: "$db-check-panel-settings-link",
     sttFailedChecksPanelSelector: "$db-check-panel-has-checks",
     checksPanelSelector: "$db-check-panel-home",
     newsPanelTitleSelector: "//span[@class='panel-title-text' and text() = 'Percona News']",
     newsPanelContentSelector: "//span[contains(text(), 'Percona News')]/ancestor::div[contains(@class, 'panel-container')]//div[contains(@class, 'view')]",
     noAccessRightsSelector: '$db-check-panel-no-access',
+    updateWidget: {
+      base: {
+        checkUpdateButton: "#refresh",
+        updateProgressModal: ".modal-content",
+        successUpgradeMsgSelector: ".modal-content",
+        lastCheckSelector: ".last-check-wrapper > p",
+        triggerUpdate: "button[ng-click='update()']",
+        reloadButtonAfterUpgrade: "button[ng-click='reloadAfterUpdate()'",
```

Comment:
Is the closing bracket missing, or is it intentional? Question is how does it work then?

Reply:
For some reason it worked actually, but fixed now. Thank you